ocs_ci/helpers/keyrotation_helper.py

```diff
@@ -0,0 +1,294 @@
+        if not self.is_keyrotation_enable():
+            log.info("Keyrotation is Already in  Disabled state.")
+            return True
+
+        param = '[{"op":"replace","path":"/spec/encryption/keyRotation/enable","value":False}]'
+        self.storagecluster_obj.patch(params=param, format_type="json")
+        self.storagecluster_obj.wait_for_resource(
+            constants.STATUS_READY,
+            self.storagecluster_obj.resource_name,
+            column="PHASE",
+            timeout=180,
+        )
+        self.storagecluster_obj.reload_data()
+        log.info("Keyrotation is Disabled in storagecluster object.")
+        return True
+
+    def is_keyrotation_enable(self):
+        """
+        Checks if key rotation is enabled for the storage cluster.
+
+        Returns:
+            bool: True if key rotation is enabled, False otherwise.
+        """
+        cmd = f" get {self.resource_name}  -o jsonpath='{{.items[*].spec.encryption.keyRotation.enable}}'"
+        cmd_out = self._exec_oc_cmd(cmd)
+        if (cmd_out is True) or (cmd_out is None):
+            log.info("Keyrotation in storagecluster object is enabled.")
+            return True
+        log.info("Keyrotation in storagecluster object is not enabled.")
+        return False
+
+
+class NoobaaKeyrotation(KeyRotation):
+    """
+    Extends KeyRotation class to handle key rotation operations for Noobaa.
+    """
+
+    def __init__(self):
+        """
+        Initializes NoobaaKeyrotation object.
+        """
+        super().__init__()
+
+    def get_noobaa_keyrotation_schedule(self):
+        """
+        Retrieves the current key rotation schedule for Noobaa.
+
+        Returns:
+            str: The key rotation schedule for Noobaa.
+
+        Raises:
+            ValueError: If the key rotation schedule is not found or is invalid.
+        """
+        cmd = " get noobaas.noobaa.io -o jsonpath='{.items[*].spec.security.kms.schedule}'"
+
+        cmd_out = self._exec_oc_cmd(cmd=cmd, out_yaml_format=False)
+        if cmd_out == "":
+            raise ValueError("Noobaa Keyrotation schedule is not found.")
+        log.info(f"Noobaa Keyrotation schedule: {cmd_out}")
+        return cmd_out.strip()
+
+    def is_noobaa_keyrotation_enable(self):
+        """
+        Checks if key rotation is enabled for Noobaa.
+
+        Returns:
+            bool: True if key rotation is enabled for Noobaa, False otherwise.
+        """
+        cmd = " get noobaas.noobaa.io -o jsonpath='{.items[*].spec.security.kms.enableKeyRotation}'"
+
+        cmd_out = self._exec_oc_cmd(cmd=cmd)
+        if cmd_out == "true":
+            log.info("Noobaa Keyrotation is Enabled.")
+            return True
+        log.info("Noobaa Keyrotation is disabled.")
+        return False
+
+    def get_noobaa_backend_secret(self):
+        """
+        Retrieves the backend secret for Noobaa.
+
+        Returns:
+            str, str: A tuple containing the Noobaa backend root key and secret.
```

Comment:
tuple (str, str): containing the Noobaa backend root key and secret

Reply:
Done